From db28c21cbe16efe4c6f0de3ef6541c1487d00348 Mon Sep 17 00:00:00 2001
From: Chris Rummel <crummel@microsoft.com>
Date: Wed, 16 Dec 2020 19:36:48 -0600
Subject: [PATCH 15/15] Don't canonicalize all ref packages to RTM versions for
 source-build. This is because we turn on BuildAllPackages in runtime since we
 don't use the previous version's prebuilt bits.  This target makes the
 version numbers mismatch, resulting in errors like error MSB3106: Assembly
 strong name
 "/home/chris/dotnet_source-build/packages/restored/system.io.pipelines/5.0.0/ref/net5.0/System.I
 O.Pipelines.dll" is either a path which could not be found or it is a full
 assembly name which is badly formed.

because while 5.0.x was restored properly, this target changes the reference in between restoring and building.
---
 eng/targets/ResolveReferences.targets | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/eng/targets/ResolveReferences.targets b/eng/targets/ResolveReferences.targets
index 032e73b069..9029bd08fc 100644
--- a/eng/targets/ResolveReferences.targets
+++ b/eng/targets/ResolveReferences.targets
@@ -259,7 +259,7 @@
     about the "invalid strong name".
   -->
   <Target Name="_UseRTMReferenceAssemblies"
-      Condition=" '$(MSBuildProjectName)' != 'RepoTasks' "
+      Condition=" '$(MSBuildProjectName)' != 'RepoTasks' and '$(DotNetBuildFromSource)' != 'true' "
       AfterTargets="ResolvePackageAssets"
       BeforeTargets="GenerateBuildDependencyFile;GeneratePublishDependencyFile;ILLink;ResolveLockFileReferences"
       DependsOnTargets="ResolvePackageAssets">
-- 
2.18.0

